internal/nix/nix.go

```diff
@@ -412,3 +413,21 @@ func parseInsecurePackagesFromExitError(errorMsg string) []string {
 
 	return insecurePackages
 }
+
+func RestartDaemon(ctx context.Context) error {
+	if runtime.GOOS != "darwin" {
+		return nil
+	}
+	err := exec.CommandContext(ctx, "launchctl", "bootout", "system", "/Library/LaunchDaemons/org.nixos.nix-daemon.plist").Run()
+	if err != nil {
+		return err
+	}
+	err = exec.CommandContext(ctx, "launchctl", "bootstrap", "system", "/Library/LaunchDaemons/org.nixos.nix-daemon.plist").Run()
+	if err != nil {
+		return err
```

Comment:
If error, should we just print error saying unable to restart daemon. Telling user they need to restart it manually to get custom cache?